app/utilities/decimal_places.py

```diff
@@ -99,7 +99,7 @@ def custom_format_unit(
     return formatted_unit
 
 
-def get_number_format(value: int | float | Decimal, locale: Locale | str) -> str:
+def get_number_format(value: int | float | Decimal, locale: Locale | str) -> Any:
```

Comment:
I changed it to `Any` because mypy was complaining that it returns `Any`, I retried it as str and mypy was complaining, maybe a `# type: ignore` is required?

Reply:
Yeah, I think it'd be better to specify the return type and then use a type ignore if the problem is an untyped method from babel - just looking at this though and it seems the number format is of type NumberPattern from babel.numbers rather than a string

Reply:
And if you make that change, mypy is going to complain in `custom_format_unit` that 'Argument "format" to "format_unit" has incompatible type "NumberPattern"; expected "str | None"'
I think babel has incorrect type hints here, and that error can be ignored, might be worth a second person verifying this too

Reply:
Correct, the babel one should be `str | NumberPattern | None`